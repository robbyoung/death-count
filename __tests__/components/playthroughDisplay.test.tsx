import React from 'react';
import renderer from 'react-test-renderer';
import PlaythroughDisplay from '../../app/components/playthroughDisplay';

describe('Playthrough Display Component', () => {
    it('will correctly render playthrough details', () => {
        const component = renderer.create(<PlaythroughDisplay
            playthrough={{
                deathCount: 80,
                gameId: '123',
                gameName: 'Test Game',
                id: '456',
                name: 'Test Playthrough',
                selected: false,
            }}
            onSelect={() => undefined}
            onDelete={() => undefined}
        />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('will correctly details of the selected playthrough', () => {
        const component = renderer.create(<PlaythroughDisplay
            playthrough={{
                deathCount: 80,
                gameId: '123',
                gameName: 'Test Game',
                id: '456',
                name: 'Test Playthrough',
                selected: true,
            }}
            onSelect={() => undefined}
            onDelete={() => undefined}
        />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});