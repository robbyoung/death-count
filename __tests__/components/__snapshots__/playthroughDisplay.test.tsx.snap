// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playthrough Display Component will correctly details of the selected playthrough 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#e84641",
      "borderRadius": 10,
      "margin": 15,
      "marginBottom": 0,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 22,
          }
        }
      >
        Test Playthrough
      </Text>
      <Text
        style={
          Object {
            "color": "#592a29",
            "fontSize": 18,
          }
        }
      >
        80
         Deaths  •  
        Test Game
      </Text>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "display": "none",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "color": undefined,
            },
            Object {
              "color": "#ffffff",
              "fontSize": 30,
              "paddingRight": 5,
            },
            Object {
              "fontFamily": "FontAwesome5FreeSolid",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`Playthrough Display Component will correctly render playthrough details 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#e84641",
      "borderRadius": 10,
      "margin": 15,
      "marginBottom": 0,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 22,
          }
        }
      >
        Test Playthrough
      </Text>
      <Text
        style={
          Object {
            "color": "#592a29",
            "fontSize": 18,
          }
        }
      >
        80
         Deaths  •  
        Test Game
      </Text>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "color": undefined,
            },
            Object {
              "color": "#ffffff",
              "fontSize": 30,
              "paddingRight": 5,
            },
            Object {
              "fontFamily": "FontAwesome5FreeSolid",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
